src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/Switch-AzureSqlDatabaseInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,137 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Services;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Switch, "AzureRmSqlDatabaseInstanceFailoverGroup", DefaultParameterSetName = SwitchIFGDefaultSet,
+        SupportsShouldProcess = true), OutputType(typeof(AzureSqlInstanceFailoverGroupModel))]
+    public class SwitchAzureSqlInstanceFailoverGroup : AzureSqlInstanceFailoverGroupCmdletBase
+    {
+        /// <summary>
+        /// Parameter set name for the default switch.
+        /// </summary>
+        private const string SwitchIFGDefaultSet = "SwitchIFGDefault";
+
+        /// <summary>
+        /// Parameter set name for switch with an Input Object.
+        /// </summary>
+        protected const string SwitchIFGByInputObjectParameterSet =
+            "Switch a Instance Failover Group from AzureSqlInstanceFailoverGroupModel instance definition";
+        
+        /// <summary>
+        /// Parameter set name for switch with a resource ID.
+        /// </summary>
+        private const string SwitchIFGByResourceIdSet = "Switch a Instance Failover Group from Resource Id";
+        
+        /// <summary>
+        /// Gets or sets the name of the InstanceFailoverGroup to use.
+        /// </summary>
+        [Parameter(ParameterSetName = SwitchIFGDefaultSet, 
+            Mandatory = true,
+            Position = 2,
+            HelpMessage = "The name of the Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+		/// Instance Failover Group object to switch
+		/// </summary>
+		[Parameter(ParameterSetName = SwitchIFGByInputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The Instance Failover Group object to switch")]
+        [ValidateNotNullOrEmpty]
+        public Model.AzureSqlInstanceFailoverGroupModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource ID of the Instance Failover Group to switch.
+        /// </summary>
+        [Parameter(ParameterSetName = SwitchIFGByResourceIdSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Resource ID of the Instance Failover Group to switch.")]
```

Comment:
the group to switch **to**? This is the new primary right? Can you use terminology like "the failover group that should become the new primary" ?